Error: The terraform-provider-azurerm_v3.84.0_x5 plugin crashed!

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

Terraform v1.6.5 on linux_amd64

### AzureRM Provider Version

3.84.0

### Affected Resource(s)/Data Source(s)

resource "azurerm_virtual_machine" "test-terraform-vm2"

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {}
}

data "azurerm_virtual_machine" "test-terraform-vm2" {
  name                = "test-terraform-vm2"
  resource_group_name = "terraform_test"
}

# imported VM details to manage
resource "azurerm_virtual_machine" "test-terraform-vm2" {
  name                = data.azurerm_virtual_machine.test-terraform-vm2.name
  resource_group_name = data.azurerm_virtual_machine.test-terraform-vm2.resource_group_name
  location            = data.azurerm_virtual_machine.test-terraform-vm2.location
  vm_size             = "Standard_B1ls"
  delete_data_disks_on_termination = false
  delete_os_disk_on_termination    = false

  lifecycle {
  ignore_changes = [
    additional_capabilities
  ]
}
}
```


### Debug Output/Panic Output

```shell
╷
│ Error: Request cancelled
│ 
│   with azurerm_virtual_machine.test-terraform-vm2,
│   on main.tf line 12, in resource "azurerm_virtual_machine" "test-terraform-vm2":
│   12: resource "azurerm_virtual_machine" "test-terraform-vm2" {
│ 
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-azurerm_v3.84.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 135 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/legacy.expandAzureRmVirtualMachineAdditionalCapabilities(0xc001952270?)
        github.com/hashicorp/terraform-provider-azurerm/internal/services/legacy/virtual_machine_resource.go:1762 +0x108
github.com/hashicorp/terraform-provider-azurerm/internal/services/legacy.resourceVirtualMachineCreateUpdate(0xc001a86400, {0x820d2e0?, 0xc00193e900})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/legacy/virtual_machine_resource.go:714 +0xd31
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x99ab090?, {0x99ab090?, 0xc001813830?}, 0xd?, {0x820d2e0?, 0xc00193e900?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:800 +0x163
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc001372000, {0x99ab090, 0xc001813830}, 0xc0015aa680, 0xc001a86080, {0x820d2e0, 0xc00193e900})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:919 +0x83a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0004aa510, {0x99ab090?, 0xc001813470?}, 0xc001cbc690)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:1060 +0xdbc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc00125d900, {0x99ab090?, 0xc001812870?}, 0xc001a89e30)
        github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:859 +0x56a
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x8d14c00?, 0xc00125d900}, {0x99ab090, 0xc001812870}, 0xc001a89dc0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:467 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002a61e0, {0x99bca60, 0xc001648680}, 0xc001332120, 0xc001a2eba0, 0xf54a9f8, 0x0)
        google.golang.org/grpc@v1.58.3/server.go:1374 +0xde7
google.golang.org/grpc.(*Server).handleStream(0xc0002a61e0, {0x99bca60, 0xc001648680}, 0xc001332120, 0x0)
        google.golang.org/grpc@v1.58.3/server.go:1751 +0x9e7
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        google.golang.org/grpc@v1.58.3/server.go:986 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 29
        google.golang.org/grpc@v1.58.3/server.go:997 +0x145

Error: The terraform-provider-azurerm_v3.84.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

The vm_size changes has applied to the vm

### Actual Behaviour

╷
│ Error: Request cancelled
│ 
│   with azurerm_virtual_machine.test-terraform-vm2,
│   on main.tf line 12, in resource "azurerm_virtual_machine" "test-terraform-vm2":
│   12: resource "azurerm_virtual_machine" "test-terraform-vm2" {
│ 
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-azurerm_v3.84.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 135 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/legacy.expandAzureRmVirtualMachineAdditionalCapabilities(0xc001952270?)
        github.com/hashicorp/terraform-provider-azurerm/internal/services/legacy/virtual_machine_resource.go:1762 +0x108
github.com/hashicorp/terraform-provider-azurerm/internal/services/legacy.resourceVirtualMachineCreateUpdate(0xc001a86400, {0x820d2e0?, 0xc00193e900})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/legacy/virtual_machine_resource.go:714 +0xd31
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x99ab090?, {0x99ab090?, 0xc001813830?}, 0xd?, {0x820d2e0?, 0xc00193e900?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:800 +0x163
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc001372000, {0x99ab090, 0xc001813830}, 0xc0015aa680, 0xc001a86080, {0x820d2e0, 0xc00193e900})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:919 +0x83a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0004aa510, {0x99ab090?, 0xc001813470?}, 0xc001cbc690)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:1060 +0xdbc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc00125d900, {0x99ab090?, 0xc001812870?}, 0xc001a89e30)
        github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:859 +0x56a
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x8d14c00?, 0xc00125d900}, {0x99ab090, 0xc001812870}, 0xc001a89dc0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:467 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002a61e0, {0x99bca60, 0xc001648680}, 0xc001332120, 0xc001a2eba0, 0xf54a9f8, 0x0)
        google.golang.org/grpc@v1.58.3/server.go:1374 +0xde7
google.golang.org/grpc.(*Server).handleStream(0xc0002a61e0, {0x99bca60, 0xc001648680}, 0xc001332120, 0x0)
        google.golang.org/grpc@v1.58.3/server.go:1751 +0x9e7
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        google.golang.org/grpc@v1.58.3/server.go:986 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 29
        google.golang.org/grpc@v1.58.3/server.go:997 +0x145

Error: The terraform-provider-azurerm_v3.84.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

### Steps to Reproduce

terraform apply

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/legacy/virtual_machine_resource.go
+++ b/internal/services/legacy/virtual_machine_resource.go
@@ -1755,7 +1755,7 @@ func expandAzureRmVirtualMachineDiagnosticsProfile(d *pluginsdk.ResourceData) *c
 
 func expandAzureRmVirtualMachineAdditionalCapabilities(d *pluginsdk.ResourceData) *compute.AdditionalCapabilities {
 	additionalCapabilities := d.Get("additional_capabilities").([]interface{})
-	if len(additionalCapabilities) == 0 {
+	if len(additionalCapabilities) == 0 || additionalCapabilities[0] == nil {
 		return nil
 	}
 

